src/assets/toolkit/scripts/lib/component/slideshow.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+import {u} from 'umbrellajs';
+export class Slideshow {
+  constructor (element, {
+    slideHolder = element.querySelector('.js-Slideshow-slides'),
+    slides = element.querySelectorAll('.js-Slideshow-slide'),
+    nextTrigger = element.querySelector('.js-Slideshow-next'),
+    prevTrigger = element.querySelector('.js-Slideshow-prev'),
+    currentCountElement = element.querySelector('.js-Slideshow-current'),
+    totalCountElement = element.querySelector('.js-Slideshow-total'),
```

Comment:
I'm curious what @erikjung thinks about my thoughts below...

I'm wondering, should the querying of the elements happen at this point? Or should this just be the CSS selector?

For example, should it be something like this?
```js
  constructor (element, {
    slideHolderSelector = '.js-Slideshow-slides',
    // rest of selectors...
  } = {}) {
    
    const  slideHolder = u(slideHolderSelector);

    // We could even make sure the slideHolder exists, since we depend on it
    if (!slideHolder) {
      throw new Error(`No element was found using the "${slideHolderSelector}" CSS selector.`);
    }

    // Now that we are sure the slideHolder exists, let's assign it. :) 
    Object.assign(this, {
      slideHolder,
      // more assignments...
    }
```